### ISSUE TITLE
Tooltip/popover doesn't work correctly when text is long and without space

### ISSUE THREAD

**Issue by Hiieu** [twbs/bootstrap#16871]
http://jsfiddle.net/19jaoah4/

It goes outside the popup.

**Comment by twbs-lmvtfy** [comment]
Hi @Hiieu!

You appear to have posted a live example (http://fiddle.jshell.net/p8d23y74/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 15, column 7: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.0.0
* line 23, column 7: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.0.0

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by kkirsche** [comment]
@Hileu Please update to the latest version of Bootstrap. and let us know if the error persists. We do not do support for old versions of Bootstrap.

**Comment by aelharrak** [comment]
Hello,

This is not a bootstrap issue it's related to the html issues :)

**Comment by jared-christensen** [comment]
Here is example of the fix I am proposing. https://jsfiddle.net/frpeqfmL/

**Comment by twbs-lmvtfy** [comment]
Hi @jared-christensen!

You appear to have posted a live example (https://fiddle.jshell.net/frpeqfmL/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 15, column 7: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.0.0
* line 23, column 7: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.0.0

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
Refs #14162

**Comment by cvrebert** [comment]
Still broken in v4: http://jsfiddle.net/z4fck0o2/3/

**Comment by fat** [comment]
i'm going to tag this as a css issue.

You css nerds just need to use `word-break` or something :+1:

**Comment by cvrebert** [comment]
CSS possibilities:
* [`word-break`](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break)
* [`word-wrap`](https://developer.mozilla.org/en-US/docs/Web/CSS/word-wrap)
* Consider using one of the above along with [`hyphens`](https://developer.mozilla.org/en-US/docs/Web/CSS/hyphens)

Non-CSS tricks:
* [`&shy;`](https://en.wikipedia.org/wiki/Soft_hyphen)
* [zero-width space](https://en.wikipedia.org/wiki/Zero-width_space)
* [`<wbr>`](http://caniuse.com/#feat=wbr-element)

**Comment by cvrebert** [comment]
@mdo Thoughts on using `word-break` or similar?

**Comment by mdo** [comment]
I'm good with `word-wrap: break-word;`, but we've got that `reset-text` mixin in there. Do we remove that property from that mixin, start passing parameters, or dupe the property?

**Comment by cvrebert** [comment]
I vote for dropping it from the mixin like we did with `font-size`.
See #18927.

**Comment by cvrebert** [comment]
Fixed in v4 via #18927.

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team
